### ISSUE TITLE
ipsw swift-dump errors on Intel

### ISSUE THREAD

**Issue by hubert3** [blacktop/ipsw#478]
### What happened?

swift-dump fails on a framework binary from the 14.5 dyld cache

```
% ipsw swift-dump /System/Volumes/Preboot/Cryptexes/OS/System/Library/dyld/dyld_shared_cache_x86_64h FMIPCore
   ⨯ failed to precache swift fields/types: failed to precache swift fields: failed to convert vmaddr: address 0x80f312d6422c not within any mappings address range
   ⨯ failed to parse swift types: failed to convert vmaddr: address 0x80f312d79920 not within any mappings address range
   ⨯ failed to parse swift protocols: failed to read swift protocol at address 0x7ffb12d6af18: failed to read signature requirement type mangled name: failed to seek to indirect context descriptor: address 0x80eb12d6f7a0 not within any mappings address range
   ⨯ failed to parse swift protocol conformances: failed to read swift protocol conformance at address 0x7ffb12d4efa0: failed to read protocol name: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor parent context: failed to read swift context descriptor pointer at address 0x80f31e60f24a: failed to Seek to address 0x80f31e60f24a: address 0x80f31e60f24a not within any mappings address range
   ⨯ failed to parse swift associated types: failed to read associated type substituted type name: failed to seek to indirect context descriptor: address 0x80e810f75320 not within any mappings address range
```

### How can we reproduce this?

Run `ipsw swift-dump /System/Volumes/Preboot/Cryptexes/OS/System/Library/dyld/dyld_shared_cache_x86_64h FMIPCore` on Sonoma 14.5 Intel

### ipsw version

```bash
Version: v3.1.482, BuildCommit: ec3f22890
```

### Search

- [X] I did search for other open and closed issues before opening this

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Additional context

Output above is running latest git code with the following diff

This change fixed swift-dump for this binary on arm64e but not on Intel

```
% git diff
diff --git a/pkg/dyld/types.go b/pkg/dyld/types.go
index 892d02996..45994371a 100644
--- a/pkg/dyld/types.go
+++ b/pkg/dyld/types.go
@@ -539,6 +539,7 @@ func (i CacheSlideInfo5) SlidePointer(ptr uint64) uint64 {
                return 0
        }
        pointer := CacheSlidePointer5(ptr)
+       if (ptr & 0xFFF0_0000_0000_0000) == 0 {return ptr}
        if pointer.Authenticated() {
                return i.ValueAdd + pointer.Value()
        }


```

**Comment by blacktop** [comment]
pushing out a release now with the fix.  please test and close if it resolves this issue.

thanks for reporting!

**Comment by hubert3** [comment]
Confirmed fixed, swift-dump now works on 14.5 FMIPCore and FMFCore on Intel with no errors 👍
